java/ql/lib/semmle/code/java/frameworks/spring/SpringWebMethod.qll

```diff
@@ -0,0 +1,30 @@
+/** Provides models of taint flow in `org.springframework.web.method` */
+
+import java
+private import semmle.code.java.dataflow.ExternalFlow
+
+// currently only models classes in the `support` subpackage
+private class FlowSummaries extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        // for review: arguably this shouldn't be modeled as the implementations of resolveArgument that I've seen are effectively sanitized
+        "org.springframework.web.method.support;HandlerMethodArgumentResolver;true;resolveArgument;;;Argument[2];ReturnValue;taint",
+        "org.springframework.web.method.support;UriComponentsContributor;true;contributeMethodArgument;;;Argument[1];Argument[2];taint",
+        "org.springframework.web.method.support;UriComponentsContributor;true;contributeMethodArgument;;;Argument[1];Argument[3];taint",
+        // InvocableHandlerMethod is not modeled as it is difficult to model method-like classes with CSV
+        // This is a very broad definition of data flow; there is a method `setRedirectModelScenario(boolean)` which is used to determine which of the `Default` and `Redirect` models are returned by `getModel`, and the methods that deal with attributes below are convenience methods for `.getMethod().*`.
```

Comment:
.getModel().*?

Reply:
e.g. `addAttribute` is a shortcut for `getModel().addAttribute`